sdk/src/azure/iot/az_iot_hub_client.c

```diff
@@ -59,13 +61,16 @@ AZ_NODISCARD az_result az_iot_hub_client_get_user_name(
 
   const az_span* const module_id = &(client->_internal.options.module_id);
   const az_span* const user_agent = &(client->_internal.options.user_agent);
+  const az_span* const model_id = &(client->_internal.options.model_id);
 
   az_span mqtt_user_name_span
       = az_span_init((uint8_t*)mqtt_user_name, (int32_t)mqtt_user_name_size);
 
   int32_t required_length = az_span_size(client->_internal.iot_hub_hostname)
       + az_span_size(client->_internal.device_id) + az_span_size(hub_service_api_version)
       + (int32_t)sizeof(hub_client_forward_slash);
+  required_length += az_span_size(*model_id) > 0 ? az_span_size(hub_service_preview_api_version)
+                                                 : az_span_size(hub_service_api_version);
   if (az_span_size(*module_id) > 0)
   {
     required_length += az_span_size(*module_id) + (int32_t)sizeof(hub_client_forward_slash);
```

Comment:
Is this the right check?  Because we url_encode later on and it can be bigger, shouldn't we canculate from what encoded length would be?

Reply:
this one specifically is determining whether we use the preview API string or the regular API string. We have to skip the calculation of the model id since we url encode it later and don't know how long it will be